src/Adapter/Product/SpecificPrice/CommandHandler/EditProductSpecificPriceHandler.php

```diff
@@ -0,0 +1,146 @@
+    {
+        $specificPrice = $this->specificPriceRepository->get($command->getSpecificPriceId());
+
+        $this->specificPriceRepository->partialUpdate(
+            $specificPrice,
+            $this->fillUpdatableProperties($command, $specificPrice)
+        );
+    }
+
+    /**
+     * @param EditProductSpecificPriceCommand $command
+     * @param SpecificPrice $specificPrice
+     *
+     * @return string[]
+     */
+    private function fillUpdatableProperties(EditProductSpecificPriceCommand $command, SpecificPrice $specificPrice): array
+    {
+        $updatableProperties = [];
+        if (null !== $command->getReduction()) {
+            $specificPrice->reduction_type = $command->getReduction()->getType();
+            $specificPrice->reduction = (float) (string) $command->getReduction()->getValue();
+            $updatableProperties = [
+                'reduction_type',
+                'reduction',
+            ];
+        }
+
+        if (null !== $command->includesTax()) {
+            $specificPrice->reduction_tax = $command->includesTax();
+            $updatableProperties[] = 'reduction_tax';
+        }
+
+        if (null !== $command->getPrice()) {
+            $specificPrice->price = (float) (string) $command->getPrice();
+            $updatableProperties[] = 'price';
+        }
+
+        if (null !== $command->getFromQuantity()) {
+            $specificPrice->from_quantity = $command->getFromQuantity();
+            $updatableProperties[] = 'from_quantity';
+        }
+
+        if (null !== $command->getShopGroupId()) {
+            $specificPrice->id_shop_group = $command->getShopGroupId()->getValue();
+            $updatableProperties[] = 'id_shop_group';
+        }
+
+        if (null !== $command->getShopId()) {
+            $specificPrice->id_shop = $command->getShopId()->getValue();
+            $updatableProperties[] = 'id_shop';
+        }
+
+        if (null !== $command->getCombinationId()) {
+            $specificPrice->id_product_attribute = $command->getCombinationId()->getValue();
+            $updatableProperties[] = 'id_product_attribute';
+        }
+
+        if (null !== $command->getCurrencyId()) {
+            $specificPrice->id_currency = $command->getCurrencyId()->getValue();
+            $updatableProperties[] = 'id_currency';
+        }
+
+        if (null !== $command->getCountryId()) {
+            $specificPrice->id_country = $command->getCountryId()->getValue();
+            $updatableProperties[] = 'id_country';
+        }
+
+        if (null !== $command->getGroupId()) {
+            $specificPrice->id_group = $command->getGroupId()->getValue();
+            $updatableProperties[] = 'id_group';
+        }
+
+        if (null !== $command->getCustomerId()) {
+            $specificPrice->id_customer = $command->getCustomerId()->getValue();
+            $updatableProperties[] = 'id_customer';
+        }
+
+        if (null !== $command->getDateTimeFrom()) {
```

Comment:
You need to add the updatable properties here, especially once this PR will be merged https://github.com/PrestaShop/PrestaShop/pull/25913

The whole partial update was buggy, it actually always saved everything, so once the bug is fixed if the updated properties are not correctly set the data won't be saved